azure-kusto-data/azure/kusto/data/helpers.py

```diff
@@ -11,10 +11,10 @@ def to_pandas_timedelta(raw_value) -> "pandas.Timedelta":
         # Kusto saves up to ticks, 1 tick == 100 nanoseconds
         return pd.to_timedelta(raw_value * 100, unit="ns")
     if isinstance(raw_value, str):
-        # The timespan format Kusto returns is 'd.hh.mm.ss.0000000'
-        # Pandas expects 'd days hh.mm.ss.0000000'
+        # The timespan format Kusto returns is 'd.hh:mm:ss.0000000' (or 'hh:mm:ss.0000000')
+        # Pandas expects 'd days hh:mm:ss.0000000'
         parts = raw_value.split(".")
-        if len(parts) < 2:
+        if len(parts) <= 2:
```

Comment:
This will ruin the conversion for timespans greater than 24 hours. 
I think I'll switch the check to use regex..

Reply:
No, it will not, as long as the "seconds" part still contains decimals.
But since the decimal part may be absent, I have pushed a further commit to address this, which could be faster than the regex.